pkg/istio-agent/xds_proxy.go

```diff
@@ -650,9 +650,7 @@ func (p *XdsProxy) buildUpstreamClientDialOpts(sa *Agent) ([]grpc.DialOption, er
 		keepaliveOption, initialWindowSizeOption, initialConnWindowSizeOption, msgSizeOption,
 	}
 
-	if !sa.secOpts.FileMountedCerts {
-		dialOptions = append(dialOptions, grpc.WithPerRPCCredentials(caclient.NewXDSTokenProvider(sa.secOpts)))
-	}
+	dialOptions = append(dialOptions, grpc.WithPerRPCCredentials(caclient.NewXDSTokenProvider(sa.secOpts)))
```

Comment:
@ramaraochavali  will this break you?

Before:
* Only mTLS used
* After: mTLS *and* token used

Reply:
I am a bit worried its actually `After: mTLS and token used and if token cannot be fetched fail hard` which may break VMs...

Reply:
If one of the authenticators pass, the client authentication will pass. After adding a token authenticator, the worst case for the authentication at the newly added token authenticator is that the authentication at the token authenticator fails, which is equivalent to before adding the token authenticator. 
Therefore, I am not sure why adding a token authenticator will break the authentication. Can you elaborate?

Reply:
I am not 100% sure it will but my concern was on the client side, not the server side. That gRPC will call the token provider which will send an error and break the whole request.

Maybe it never returns an error , would need to double check.

Reply:
Thanks for the clarification. Based on the review comment that the token authentication must be used for XDS, attaching a token on the client side is inevitable.

Reply:
Thanks Lei. I did some testing and verified that we never return an error on the token provider - even if we cannot read the token. Ity would be great if you can make this change so that we are testing this:
```
diff --git a/pkg/istio-agent/agent_test.go b/pkg/istio-agent/agent_test.go
index 84aeafae1b..e80e469fd5 100644
--- a/pkg/istio-agent/agent_test.go
+++ b/pkg/istio-agent/agent_test.go
@@ -154,6 +154,7 @@ func TestAgent(t *testing.T) {
                        a.Security.CertChainFilePath = cfg.CertificatePath
                        a.Security.KeyFilePath = cfg.PrivateKeyPath
                        a.Security.RootCertFilePath = cfg.CaCertificatePath
+                       a.Security.JWTPath = "bogus"
                        a.ProxyConfig.ProxyMetadata = map[string]string{}
                        a.ProxyConfig.ProxyMetadata[MetadataClientCertChain] = filepath.Join(dir, "cert-chain.pem")
                        a.ProxyConfig.ProxyMetadata[MetadataClientCertKey] = filepath.Join(dir, "key.pem")
```

Since right now the test has a valid JWT token so its not ideal test.

Reply:
A new test case with (a.Security.JWTPath = "bogus") has been added.